unity/kuka-ar-unity/Assets/Project/Scripts/EventSystem/Controllers/Menu/TopMenuController.cs

```diff
@@ -44,57 +22,29 @@ private void Awake()
 
         private void Start()
         {
+            var topMenu = GetComponent<Transform>().gameObject;
             var constantTopPanel = topMenu.GetComponent<RectTransform>()
                 .Find("ConstantInfoPanel").GetComponent<RectTransform>().gameObject;
             
-            toolNo = constantTopPanel.transform.Find("ToolBean").GetComponent<RectTransform>().gameObject
+            tool = constantTopPanel.transform.Find("ToolBean").GetComponent<RectTransform>().gameObject
+                .transform.Find("ToolBackground").GetComponent<RectTransform>().gameObject
                 .transform.Find("Tool").GetComponent<TMP_Text>();
-            baseNo = constantTopPanel.transform.Find("WorldBean").GetComponent<RectTransform>().gameObject
+            world = constantTopPanel.transform.Find("WorldBean").GetComponent<RectTransform>().gameObject
+                .transform.Find("WorldBackground").GetComponent<RectTransform>().gameObject
                 .transform.Find("World").GetComponent<TMP_Text>();
-
-            ConstantPanelState = LogicStates.Waiting;
-            CoordListState = LogicStates.Waiting;
-            
-            TopMenuEvents.Events.OnBeanClick += OnToolOrBaseClick;
-            TopMenuEvents.Events.OnDragTopMenuSlider += OnMenuDrag;
-            TopMenuEvents.Events.OnDropTopMenuSlider += OnMenuDrop;
             
             trackedRobot.ActiveBaseUpdated += OnBaseValueChange;
             trackedRobot.ActiveToolUpdated += OnToolValueChange;
         }
 
-        private void OnToolOrBaseClick(int uid)
-        {
-            if (id == uid)
-            {
-                ConstantPanelState = LogicStates.Running;
-            }
-        }
-
-        private void OnMenuDrag(int uid)
-        {
-            if (id == uid)
-            {
-                CoordListState = LogicStates.Running;
-            }
-        }
-
-        private void OnMenuDrop(int uid)
-        {
-            if (id == uid)
-            {
-                CoordListState = LogicStates.AutoPulling;
-            }
-        }
-
         private void OnBaseValueChange(object _, KRLInt e)
         {
-            baseNo.text = "Base\n" + e.Value;
+            world.text = "World\n" + e.Value;
```

Comment:
same here

Reply:
## After review fix 
All of this point has been fix in 82441b478fb91e017afd338742a38d835f7da170